optuna/storages/_rdb/storage.py

```diff
@@ -464,9 +464,46 @@ def _create_new_trial(
 
         session = self.scoped_session()
 
-        # Ensure that that study exists.
-        models.StudyModel.find_or_raise_by_id(study_id, session)
+        try:
```

Comment:
Memo: Try n (maybe 3) times and propagate the `OperationalError` in case they all fail. It should be easier to debug.